test/infamy/netconf.py

```diff
@@ -77,40 +79,54 @@ def __init__(self,
         self.mapping = mapping
         self.ly = libyang.Context(yangdir)
         self._ncc_init(location)
+        #self.ncc._fetch_connection_ip()
+        #self.ncc._debug()
+
         self.modules = {}
         self._ly_bootstrap(yangdir)
 
         del self.ly
         self.ly = libyang.Context(yangdir)
         self._ly_init(yangdir)
-        self.ncc.dispatch('<factory-default xmlns="urn:infix:factory-default:ns:yang:1.0"/>')
+        try:
+            self.ncc.dispatch('<factory-default xmlns="urn:infix:factory-default:ns:yang:1.0"/>')
+        except RpcError as err:
+            print(f"Failed sending factory-default RPC: {err}")
+            sys.exit(1)
```

Comment:
What's the reasoning here? Won't the test terminate anyway, as the exception bubbles upwards?

Reply:
Does not seems to do that, this i expected to fail the test.
ok 1 - Initialize
# [{'name': 'USB', 'class': 'infix-hardware:usb', 'state': {'admin-state': 'locked'}}, {'name': 'USB2', 'class': 'infix-hardware:usb', 'state': {'admin-state': 'locked'}}]
# Failed sending edit-config RPC: Internal error  Retrying ...
# Failed sending edit-config RPC: Internal error  Retrying ...
# Failed sending edit-config RPC: Internal error  Retrying ...
ok 2 - Lock USB ports

Reply:
I had some real trouble connecting at times (due to interfaces being lost inside containers) so I added these mostly for debugging.

Can drop it if you want?

Reply:
Replacing `sys.exit(1)` with `raise err` as discussed AFK.